pandas/tests/frame/test_block_internals.py

```diff
@@ -658,3 +658,26 @@ def test_update_inplace_sets_valid_block_values():
 
     # smoketest for OP bug from GH#35731
     assert df.isnull().sum().sum() == 0
+
+
+def test_nonconsolidated_item_cache_take():
```

Comment:
i would put in pandas/tests/frame/test_indexing.py (or similar) rather than here.

Reply:
It's not really an indexing bug, but rather a bug in `take` in combination with indexing before and afterwards (and it's the assign operation that only makes the bug visible), and is specific to the consolidation state of the original dataframe. So it's more a bug in the internal machinery of consolidation / cache.